AuthorizationAndAuthentication

ltanitanil opened the pull request:
Fixes #23

Commit:
Updated to .NET Core 3.0 Preview 8

Commit:
Added authentication and authorization for WebApi.

Commit:
Added authentication and authorization for Client.

Tiktack commented on src/Blazor.Server.BusinessLayer/Services/AccountService/AccountService.cs:
```diff
@@ -0,0 +1,64 @@
+        {
+            var genderEnum = gender switch
+            {
+                "male" => Gender.Male,
```
Gender in params should be Enum. It's responsibility of class that call this method

Tiktack commented on src/Blazor.Server.BusinessLayer/Services/AccountService/IAccountService.cs:
```diff
@@ -0,0 +1,13 @@
+{
+    public interface IAccountService
+    {
+        Task Register(string email, string password, DateTime dateOfBirth, string gender, string userInformation);
```
Maybe create entity for business layer with this properties?

Tiktack commented on src/Blazor.Server.DataAccessLayer/Blazor.Server.DataAccessLayer.csproj:
```diff
@@ -5,7 +5,14 @@
 
+  <ItemGroup>
+    <Reference Include="Microsoft.Extensions.Identity.Stores">
+      <HintPath>..\..\..\..\..\..\..\Program Files\dotnet\packs\Microsoft.AspNetCore.App.Ref\3.0.0-preview8.19405.7\ref\netcoreapp3.0\Microsoft.Extensions.Identity.Stores.dll</HintPath>
```
install this from nuget, not local path

Tiktack commented on src/Blazor.Server.BusinessLayer/Blazor.Server.BusinessLayer.csproj:
```diff
@@ -2,10 +2,20 @@
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="Microsoft.IdentityModel.Tokens">
```
delete this local paths

Tiktack commented on src/Blazor.Server.WebApi/Controllers/Api/TorrentsController.cs:
```diff
@@ -24,7 +23,7 @@ public TorrentsController(IMapper mapper, ITorrentsService torrentsViewModelServ
         public async Task<TorrentsViewModel> GetTorrents(int pageIndex, SearchAndFilterCriteria criteria)
         {
-            var itemsPerPage = Constants.ITEMS_PER_PAGE;
+            const int itemsPerPage = Constants.ITEMS_PER_PAGE;
```
for what? Just use inline

Tiktack commented on src/Blazor.Shared.Models/Blazor.Shared.Models.csproj:
```diff
@@ -5,4 +5,10 @@
 
+  <ItemGroup>
+    <Reference Include="System.ComponentModel.Annotations">
+      <HintPath>..\..\..\..\..\..\..\Program Files\dotnet\sdk\NuGetFallbackFolder\system.componentmodel.annotations\4.5.0\ref\netstandard2.0\System.ComponentModel.Annotations.dll</HintPath>
```
del pls

Commit:
Deleted local paths.

Commit:
Code refactoring:
- created an entity for the business level("RegistrationViewModel");
-made some corrections...

Commit:
Deleted unnecessary namespace

Commit:
Added MatRadioButton for Gender.
